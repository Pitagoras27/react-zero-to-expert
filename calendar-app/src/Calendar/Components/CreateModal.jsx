import { addHours } from "date-fns";
import es from "date-fns/locale/es";
import DatePicker from "react-datepicker";
import Modal from "react-modal";

import "react-datepicker/dist/react-datepicker.css";
import { useModalForm, useUiStore } from "../../hooks";

const customStyles = {
  content: {
    top: "50%",
    left: "50%",
    right: "auto",
    bottom: "auto",
    marginRight: "-50%",
    transform: "translate(-50%, -50%)",
  },
};

Modal.setAppElement("#root");

export const CreateModal = () => {
  const { isDateModalOpen, onCloseModal } = useUiStore();
  const {
    title,
    notes,
    start,
    end,
    validationText,
    inputRef,
    onValueChange,
    onDateSelected,
    onSubmitEvent,
  } = useModalForm({
    title: "Carlor",
    notes: "Purification",
    start: new Date(),
    end: addHours(new Date(), 2),
  });

  return (
    <Modal
      isOpen={isDateModalOpen}
      onRequestClose={onCloseModal}
      style={customStyles}
      classModal="modal"
      overClassName="modal-fondo"
      closeTimeoutMS={200}
    >
      <h1> Nuevo evento </h1>
      <hr />
      <form className="container" onSubmit={onSubmitEvent}>
        <div className="form-group mb-2">
          <label>Fecha y hora inicio</label>
          <DatePicker
            selected={start}
            onChange={(date) => onDateSelected(date, "start")}
            className="form-control"
            dateFormat="MMMM d, yyyy h:mm aa"
            locale={es}
            timeCaption="hora"
            showTimeSelect
          />
        </div>

        <div className="form-group mb-2">
          <label>Fecha y hora fin</label>
          <DatePicker
            selected={end}
            onChange={(date) => onDateSelected(date, "end")}
            className="form-control"
            dateFormat="MMMM d, yyyy h:mm aa"
            minDate={start}
            locale={es}
            timeCaption="hora"
            showTimeSelect
          />
        </div>

        <hr />
        <div className="form-group mb-2">
          <label>Titulo y notas</label>
          <input
            type="text"
            className={`form-control ${validationText}`}
            placeholder="Título del evento"
            ref={inputRef}
            name="title"
            autoComplete="off"
            value={title}
            onChange={onValueChange}
          />
          <small id="emailHelp" className="form-text text-muted">
            Una descripción corta
          </small>
        </div>

        <div className="form-group mb-2">
          <textarea
            type="text"
            className="form-control"
            placeholder="Notas"
            rows="5"
            name="notes"
            value={notes}
            onChange={onValueChange}
          ></textarea>
          <small id="emailHelp" className="form-text text-muted">
            Información adicional
          </small>
        </div>

        <button type="submit" className="btn btn-outline-primary btn-block">
          <i className="far fa-save"></i>
          <span> Guardar</span>
        </button>
      </form>
    </Modal>
  );
};
