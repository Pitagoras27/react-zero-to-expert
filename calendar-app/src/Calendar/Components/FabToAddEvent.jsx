import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks";

export const FabToAddEvent = () => {
  const { onModalOpen } = useUiStore();
  const { onSetActiveEvent } = useCalendarStore();

  const handleAddNewEvent = () => {
    onSetActiveEvent({
      title: "",
      notes: "",
      start: new Date(),
      end: addHours(new Date(), 2),
      bgColor: "#fafafa",
      user: {
        _id: "",
        name: "",
      },
    });
    onModalOpen();
  };

  return (
    <button className="fab btn btn-primary" onClick={handleAddNewEvent}>
      <i className="fas fa-plus"></i>
    </button>
  );
};
