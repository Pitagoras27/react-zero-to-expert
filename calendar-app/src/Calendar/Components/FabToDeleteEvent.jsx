import { useCalendarStore, useUiStore } from "../../hooks";

export const FabToDeleteEvent = () => {

  const { hasActiveEvent, startDeleteEvent } = useCalendarStore();
  const { isDateModalOpen } = useUiStore();

  const handleDeleteEvent = () => {
    startDeleteEvent();
  };

  return (
    <button
      aria-label="delete-event"
      className="fab btn btn-danger fab-danger"
      onClick={handleDeleteEvent}
      style={{ display: (hasActiveEvent && !isDateModalOpen) ? '' : 'none'}}
    >
      <i className="fas fa-trash-alt"></i>
    </button>
  );
};
