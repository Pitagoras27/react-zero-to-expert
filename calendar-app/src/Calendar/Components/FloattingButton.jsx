import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks";

export const FloattingButton = () => {
  const { onModalOpen } = useUiStore();
  const { onSetActiveEvent } = useCalendarStore();

  const onAddNewEvent = () => {
    onModalOpen();
    onSetActiveEvent({
      title: "",
      notes: "",
      start: new Date(),
      end: addHours(new Date(), 2),
      bgColor: "#fafafa",
      user: {
        _id: "",
        name: "",
      },
    });
  };

  return (
    <button className="fab btn btn-primary" onClick={onAddNewEvent}>
      <i className="fas fa-plus"></i>
    </button>
  );
};
