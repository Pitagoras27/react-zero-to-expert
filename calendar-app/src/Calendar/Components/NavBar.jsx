import { useAuthStore } from "../../hooks";

export const NavBar = () => {
  const { startLogout } = useAuthStore();
  return (
    <div className='navbar navbar-dark bg-black mb-4 px-4'>
      <span className="navbar-brand">
        <i className="fas fa-calendar-alt"></i>&nbsp;
        Carlos
      </span>

      <button
        className="btn btn-outline-danger"
        onClick={startLogout}
      >
        <i className="fas fa-sign-out-alt"></i>
        <span style={{ marginLeft: '10px'}}>Logout</span>
      </button>
    </div>
  )
}