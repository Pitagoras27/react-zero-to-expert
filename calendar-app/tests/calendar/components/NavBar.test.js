import { render } from "@testing-library/react";
import { NavBar } from "../../../src/Calendar/Components/NavBar";

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => (fn) => fn(),
  useSelector: () => (fn) => fn(),
}));

describe('Test of Navbar component', () => {
  test('should render NavBar corretly', () => {
    const { container } = render(<NavBar />);

    expect(container).toMatchSnapshot();
  })
})