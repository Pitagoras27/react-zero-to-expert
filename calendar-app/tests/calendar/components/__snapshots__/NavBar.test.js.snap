// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Navbar component should render NavBar corretly 1`] = `
<div>
  <div
    class="navbar navbar-dark bg-black mb-4 px-4"
  >
    <span
      class="navbar-brand"
    >
      <i
        class="fas fa-calendar-alt"
      />
      Â  Carlos
    </span>
    <button
      class="btn btn-outline-danger"
    >
      <i
        class="fas fa-sign-out-alt"
      />
      <span
        style="margin-left: 10px;"
      >
        Logout
      </span>
    </button>
  </div>
</div>
`;
